### ISSUE TITLE
Generating report for multi-module project with flavors

### ISSUE THREAD

**Issue by IlyaPavlik** [Kotlin/kotlinx-kover#314]
**Describe the bug**
Build failed when I try to generate html report for multi module project with flavors.
`./gradlew clean :app:koverHtmlReportDemoDebug`

**Errors**
```
Could not determine the dependencies of task ':app:koverHtmlReportDemoDebug'.
> Could not resolve all task dependencies for configuration ':app:koverAggregatedSetupDemoDebug'.
   > Could not resolve project :libModule.
     Required by:
         project :app
      > The consumer was configured to find attribute 'kotlinx.kover.setup.name' with value 'demoDebug'. However we cannot choose between the following variants of project :libModule:
          - debugApiElements
          - debugRuntimeElements
          - releaseApiElements
          - releaseRuntimeElements
        All of them match the consumer attributes:
          - Variant 'debugApiElements' capability JaCoCoExample:libModule:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.4.2' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.setup.name (required 'demoDebug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'debugRuntimeElements' capability JaCoCoExample:libModule:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.4.2' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'debug' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'debug' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.setup.name (required 'demoDebug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'releaseApiElements' capability JaCoCoExample:libModule:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.4.2' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.setup.name (required 'demoDebug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-api' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
          - Variant 'releaseRuntimeElements' capability JaCoCoExample:libModule:unspecified:
              - Unmatched attributes:
                  - Provides com.android.build.api.attributes.AgpVersionAttr '7.4.2' but the consumer didn't ask for it
                  - Provides com.android.build.api.attributes.BuildTypeAttr 'release' but the consumer didn't ask for it
                  - Provides com.android.build.gradle.internal.attributes.VariantAttr 'release' but the consumer didn't ask for it
                  - Doesn't say anything about kotlinx.kover.setup.name (required 'demoDebug')
                  - Provides org.gradle.category 'library' but the consumer didn't ask for it
                  - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
                  - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
                  - Provides org.jetbrains.kotlin.platform.type 'androidJvm' but the consumer didn't ask for it
        The following variants were also considered but didn't match the requested attributes:
          - Variant 'koverSetupDebug' capability JaCoCoExample:libModule:unspecified:
              - Incompatible because this component declares attribute 'kotlinx.kover.setup.name' with value 'debug' and the consumer needed attribute 'kotlinx.kover.setup.name' with value 'demoDebug'
          - Variant 'koverSetupRelease' capability JaCoCoExample:libModule:unspecified:
              - Incompatible because this component declares attribute 'kotlinx.kover.setup.name' with value 'release' and the consumer needed attribute 'kotlinx.kover.setup.name' with value 'demoDebug'
```

**Expected behavior**
Should generate merged html report

**Reproducer**
app module
```
plugins {
    id 'org.jetbrains.kotlinx.kover' version '0.7.0-Alpha'
}

flavorDimensions "version"
    productFlavors {
        demo {
            dimension "version"
        }
        full {
            dimension "version"
        }
    }

dependencies {
    kover project(":libModule")
}
```

libModule
```
plugins {
    id 'org.jetbrains.kotlinx.kover' version '0.7.0-Alpha'
}
```

**Environment**
 - Kover Gradle Plugin version: 0.7.0-Alpha
 - Gradle version: gradle-7.5
 - Kotlin project type: Kotlin/Android
 - Coverage Engine version (if customized in build script): DefaultJacocoEngine
 - Other context important for this bug: -

**Comment by wezley98** [comment]
We are also seeing this for our project in `0.7.0-Alpha`, Most of modules just use `debug` and `release` but one module has `productFlavours` for `uat`,`staging`,`prod`

This one module `:config` with flavours can't be merged with the others modules.

I've also tried `missingDimensionStrategy("environment", listOf("uat", "staging", "prod"))` in the `config` module but still get the following error:

```
Could not resolve all task dependencies for configuration ':kover:koverAggregatedSetupDebug'.
   > Could not resolve project :config.
     Required by:
         project :kover
      > The consumer was configured to find attribute 'kotlinx.kover.setup.name' with value 'debug'. However we cannot choose between the following variants of project :config:
          - prodDebugApiElements
          - prodDebugRuntimeElements
          - prodReleaseApiElements
          - prodReleaseRuntimeElements
          - stagingDebugApiElements
          - stagingDebugRuntimeElements
          - stagingReleaseApiElements
          - stagingReleaseRuntimeElements
          - uatDebugApiElements
          - uatDebugRuntimeElements
          - uatReleaseApiElements
          - uatReleaseRuntimeElements
   ```

**Comment by shanshin** [comment]
Hi,
unfortunately, the alpha version supports only multi-module projects that have completely identical build variants.

**Comment by wezley98** [comment]
@shanshin That's fine, will this be a future feature?

**Comment by shanshin** [comment]
I think it will be possible to implement this.

Now we are collecting feedback and the most common use cases.

**Comment by shanshin** [comment]
Fixed in `0.7.0-Beta`
Please note that the DSL has been modified compared to `Alpha` version.
